optaweb-vehicle-routing-backend/src/test/java/org/optaweb/vehiclerouting/plugin/planner/weight/DepotAngleCustomerDifficultyWeightFactoryTest.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.vehiclerouting.plugin.planner.weight;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.optaweb.vehiclerouting.plugin.planner.SolutionFactory;
+import org.optaweb.vehiclerouting.plugin.planner.VehicleRoutingSolution;
+import org.optaweb.vehiclerouting.plugin.planner.domain.PlanningDepot;
+import org.optaweb.vehiclerouting.plugin.planner.domain.PlanningLocation;
+import org.optaweb.vehiclerouting.plugin.planner.domain.PlanningVehicle;
+import org.optaweb.vehiclerouting.plugin.planner.domain.PlanningVisit;
+import org.optaweb.vehiclerouting.plugin.planner.domain.Standstill;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.WARN)
+class DepotAngleCustomerDifficultyWeightFactoryTest {
+
+    private final PlanningLocation location1 = new PlanningLocation(1, 1.0, 1.0);
+    private final PlanningLocation location2 = new PlanningLocation(2, 1.0, 1.0);
+    private final PlanningLocation location3 = new PlanningLocation(3, 1.0, 50.0);
+
+    @InjectMocks
+    private DepotAngleCustomerDifficultyWeightFactory  weightFactory;
+
+
+    @Test
+    void createSorterWeight_close_customer_should_have_smaller_weight() {
+        VehicleRoutingSolution solution = createSolution(location1, location2, location3);
+        DepotAngleCustomerDifficultyWeightFactory.DepotAngleCustomerDifficultyWeight closeCustomerWeight =
+                weightFactory.createSorterWeight(solution,
+                solution.getVisitList().get(0));
+        DepotAngleCustomerDifficultyWeightFactory.DepotAngleCustomerDifficultyWeight farCustomerWeight =
+                weightFactory.createSorterWeight(solution,
+                    solution.getVisitList().get(1));
+        assert(closeCustomerWeight.compareTo(farCustomerWeight) < 0);
```

Comment:
The `assert` keyword is for [Java Assertions](https://docs.oracle.com/javase/8/docs/technotes/guides/language/assert.html). Java Assertions are used to check precondtions, postconditions, invariants, parameter checking, etc. and are *disabled by default*.

In tests we use the `assertThat()` method from [AssertJ](https://assertj.github.io/doc/#overview-what-is-assertj). It has a type-sensitive fluent API and gives nice error descriptions.